# Data Extraction - Airbnb Dataset

## Objective
Load the raw Airbnb dataset and perform initial inspection to understand its structure, size, and basic characteristics.


In [4]:
import pandas as pd
import numpy as np

In [5]:
file_path = "/content/airbnb_listing.csv"

df = pd.read_csv(file_path)
df.head()

/tmp/ipykernel_13062/4240784688.py:3: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(file_path)


,calculated_host_listings_count_private_rooms,host_response_rate,host_total_listings_count,review_scores_accuracy,license,calendar_last_scraped,picture_url,host_has_profile_pic,minimum_maximum_nights,host_acceptance_rate,...,host_thumbnail_url,number_of_reviews_ltm,review_scores_value,scrape_id,availability_30,host_about,beds,number_of_reviews_l30d,id,availability_365
0,0.0,NaN,NaN,3.44,NaN,2026-02-15,https://a0.muscache.com/pictures/44627226/0e72...,t,1125.0,NaN,...,NaN,0.0,3.67,20260215064701,0.0,I am a real down to earth & cool person.,2.0,0.0,2992450,321.0
1,0.0,NaN,NaN,4.88,NaN,2026-02-15,https://a0.muscache.com/pictures/prohost-api/H...,t,1125.0,NaN,...,NaN,10.0,4.78,20260215064701,0.0,"Hello! I’m a proud resident of Albany, NY, whe...",1.0,2.0,3820211,301.0
2,1.0,NaN,NaN,4.61,NaN,2026-02-15,https://a0.muscache.com/pictures/b3fc42f3-6e5e...,t,1125.0,NaN,...,NaN,20.0,4.63,20260215064701,18.0,I am an Albany native .I have lived in Ireland...,2.0,2.0,5651579,35.0
3,0.0,NaN,NaN,NaN,NaN,2026-02-15,https://a0.muscache.com/pictures/hosting/Hosti...,t,365.0,NaN,...,NaN,0.0,NaN,20260215064701,20.0,NaN,3.0,0.0,6015313,324.0
4,0.0,NaN,NaN,4.83,NaN,2026-02-15,https://a0.muscache.com/pictures/prohost-api/H...,t,1125.0,NaN,...,NaN,1.0,4.72,20260215064701,13.0,"Hello! I’m a proud resident of Albany, NY, whe...",1.0,0.0,6623339,348.0


In [6]:
df.shape

(42903, 79)

In [7]:
df.columns.tolist()

['calculated_host_listings_count_private_rooms',
 'host_response_rate',
 'host_total_listings_count',
 'review_scores_accuracy',
 'license',
 'calendar_last_scraped',
 'picture_url',
 'host_has_profile_pic',
 'minimum_maximum_nights',
 'host_acceptance_rate',
 'availability_90',
 'last_scraped',
 'neighbourhood_cleansed',
 'last_review',
 'availability_60',
 'minimum_nights_avg_ntm',
 'bedrooms',
 'review_scores_cleanliness',
 'host_response_time',
 'reviews_per_month',
 'listing_url',
 'availability_eoy',
 'amenities',
 'calculated_host_listings_count_shared_rooms',
 'host_listings_count',
 'longitude',
 'host_name',
 'host_verifications',
 'neighbourhood_group_cleansed',
 'host_picture_url',
 'room_type',
 'minimum_minimum_nights',
 'bathrooms',
 'review_scores_rating',
 'maximum_nights_avg_ntm',
 'estimated_occupancy_l365d',
 'estimated_revenue_l365d',
 'calculated_host_listings_count',
 'host_neighbourhood',
 'neighborhood_overview',
 'maximum_minimum_nights',
 'property_type',
 'n

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42903 entries, 0 to 42902
Data columns (total 79 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   calculated_host_listings_count_private_rooms  42902 non-null  float64
 1   host_response_rate                            32294 non-null  object 
 2   host_total_listings_count                     42423 non-null  float64
 3   review_scores_accuracy                        35783 non-null  float64
 4   license                                       25768 non-null  object 
 5   calendar_last_scraped                         42902 non-null  object 
 6   picture_url                                   42903 non-null  object 
 7   host_has_profile_pic                          42876 non-null  object 
 8   minimum_maximum_nights                        42889 non-null  float64
 9   host_acceptance_rate                          35340 non-null 

In [9]:
df.describe()

,calculated_host_listings_count_private_rooms,host_total_listings_count,review_scores_accuracy,minimum_maximum_nights,availability_90,availability_60,minimum_nights_avg_ntm,bedrooms,review_scores_cleanliness,reviews_per_month,...,review_scores_location,review_scores_communication,number_of_reviews_ltm,review_scores_value,scrape_id,availability_30,beds,number_of_reviews_l30d,id,availability_365
count,42902.000000,42423.000000,35783.000000,4.288900e+04,42902.000000,42902.000000,42902.000000,41363.000000,35781.000000,35783.000000,...,35779.000000,35782.000000,42902.000000,35778.000000,4.290300e+04,42902.000000,32977.000000,42902.000000,4.290300e+04,42902.000000
mean,1.500000,34.426490,4.801078,1.007491e+05,45.216237,27.189875,7.733546,1.506298,4.756458,1.357060,...,4.726750,4.857957,10.251410,4.691585,2.025102e+13,11.109249,2.038451,0.870146,5.449050e+17,195.303762
std,5.677706,164.378849,0.346241,1.466448e+07,34.015698,22.122010,31.178100,1.274222,0.377259,1.940987,...,0.375201,0.322518,21.362045,0.384735,9.498704e+08,10.771320,1.789041,1.980074,5.650362e+17,139.875467
min,0.000000,0.000000,1.000000,1.000000e+00,0.000000,0.000000,1.000000,0.000000,0.000000,0.010000,...,0.000000,0.000000,0.000000,0.000000,2.025091e+13,0.000000,0.000000,0.000000,2.726200e+04,0.000000
25%,0.000000,1.000000,4.750000,9.000000e+01,1.000000,0.000000,1.000000,1.000000,4.680000,0.240000,...,4.640000,4.850000,0.000000,4.600000,2.025092e+13,0.000000,1.000000,0.000000,3.004149e+07,39.250000
50%,0.000000,4.000000,4.900000,3.650000e+02,54.000000,28.000000,2.000000,1.000000,4.870000,0.660000,...,4.830000,4.960000,2.000000,4.780000,2.025093e+13,8.000000,2.000000,0.000000,5.934000e+17,234.000000
75%,0.000000,16.000000,5.000000,1.125000e+03,76.000000,48.000000,3.000000,2.000000,5.000000,1.850000,...,4.970000,5.000000,11.000000,4.910000,2.025093e+13,21.000000,2.000000,1.000000,1.099529e+18,334.000000
max,193.000000,8774.000000,5.000000,2.147484e+09,90.000000,60.000000,1115.000000,50.000000,5.000000,99.420000,...,5.000000,5.000000,1304.000000,5.000000,2.026022e+13,30.000000,50.000000,89.000000,1.611096e+18,365.000000


In [10]:
# count missing values in each column

df.isnull().sum().sort_values(ascending=False)

,0
neighbourhood_group_cleansed,42903
calendar_updated,42903
host_neighbourhood,26987
neighborhood_overview,20792
neighbourhood,20792
...,...
host_id,0
name,0
host_url,0
scrape_id,0


In [11]:
df.duplicated().sum()

np.int64(0)

## Column Categorization

### 1. Target Variables (KPIs)
- estimated_revenue_l365d
- estimated_occupancy_l365d

### 2. Key Features for Analysis
- room_type
- price
- accommodates
- bedrooms, bathrooms, beds
- amenities
- neighbourhood_cleansed
- host_is_superhost

### 3. Columns with High Missing Values
- host_neighbourhood
- neighborhood_overview
- host_about
- license

### 4. Columns to Drop (Completely Missing)
- neighbourhood_group_cleansed
- calendar_updated

### 5. Columns Requiring Type Conversion
- price (object → float)
- host_response_rate (% → float)
- host_acceptance_rate (% → float)
- date columns → datetime

## Conclusion

- Dataset contains 42,903 listings with 79 features
- No duplicate rows found
- Several columns have missing values, including some completely empty columns
- Key variables for business problem identified
- Data type inconsistencies observed in price and percentage fields

